src/Microsoft.Extensions.SecretManager.Tools/Internal/ProjectIdResolver.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    public class ProjectIdResolver : IDisposable
+    {
+        private const string TargetsFileName = "FindUserSecretsProperty.targets";
+        private readonly ILogger _logger;
+        private readonly string _workingDirectory;
+        private Stack<Action> _disposables = new Stack<Action>();
```

Comment:
I prefer the flexiblity thie current structure lends to the code. Also, it's CLI tool and the perf gain is so minimal it's not noticeable, may not even be measurable compared to how slow invoking a new MSBuild process will be. 

But not worth spending more cycles on debating small stuff like this. We have bigger fish to fry. Will update.